tests/it/scargo_test/it_scargo_use_cases.py

```diff
@@ -160,3 +162,35 @@ def test_build_fail_with_incorrect_dependencies():
 
     build_command_result = runner.invoke(cli, ["build"])
     assert build_command_result.exit_code != 0
+
+
+@pytest.mark.parametrize("target", TARGET)
+def test_gen_with_check_and_test(target):
+    runner = CliRunner()
+    lib_name = "test_library"
+    src_dir_path = Path("src")
+    main_dir_path = Path("main")
+    new_command_result = runner.invoke(
+        cli,
+        [
+            "new",
+            pytest.new_test_project_name,
+            f"--lib={lib_name}",
+            f"--target={target}",
+        ],
+    )
+    assert new_command_result.exit_code == 0
+
+    if target == "esp32":
```

Comment:
No if in test please. This dir name is available in [Config.project.target.source_dir](https://github.com/Spyro-Soft/scargo/blob/develop/scargo/scargo_src/sc_config.py#L55)

Reply:
removed if and used sc.config